explorer/common/trace_stream.h

```diff
@@ -126,6 +176,30 @@ class SetProgramPhase {
   ProgramPhase initial_phase_;
 };
 
+// This is a RAII class to set the source location in trace stream, destructor
+// invocation restores the initial source location.
+class SetFileContext {
+ public:
+  explicit SetFileContext(TraceStream& trace_stream,
+                          std::optional<SourceLocation> source_loc)
+      : trace_stream_(trace_stream),
+        initial_source_loc_(trace_stream.source_loc()) {
+    trace_stream_.set_source_loc(source_loc);
+  }
+
+  // This can be used for cases when source location needs to be updated
+  // multiple times within the same scope.
+  auto update_source_loc(std::optional<SourceLocation> source_loc) {
+    trace_stream_.set_source_loc(source_loc);
+  }
+
+  ~SetFileContext() { trace_stream_.set_source_loc(initial_source_loc_); }
```

Comment:
Please order the destructor after the constructor and before `update_source_loc` (and also for `SetProgramPhase`). Carbon's C++ style guide is based on the Google C++ style guide; see the [relevant section](https://google.github.io/styleguide/cppguide.html#Declaration_Order) on declaration order in classes.

Reply:
Okay